File: app/src/main/java/com/sstechcanada/todo/activities/BillingClientSetup.java
package com.sstechcanada.todo.activities;

import android.content.Context;

import com.android.billingclient.api.BillingClient;
import com.android.billingclient.api.PurchasesUpdatedListener;

public class BillingClientSetup {

    private  static BillingClient instance;

    public static BillingClient getInstance(Context context, PurchasesUpdatedListener listener){
        return instance==null ? setupBillingClient(context,listener):instance;
    }

    private static BillingClient setupBillingClient(Context context, PurchasesUpdatedListener listener) {

        BillingClient billingClient = BillingClient.newBuilder(context)
                .setListener(listener)
                .enablePendingPurchases()
                .build();

        return billingClient;
    }
}

File: app/src/main/java/com/sstechcanada/todo/activities/BillingClientSetup.kt
package com.sstechcanada.todo.activities

import android.content.Context
import com.android.billingclient.api.BillingClient
import com.android.billingclient.api.PurchasesUpdatedListener

object BillingClientSetup {
    private val instance: BillingClient? = null
    fun getInstance(context: Context, listener: PurchasesUpdatedListener): BillingClient {
        return instance ?: setupBillingClient(context, listener)
    }

    private fun setupBillingClient(
        context: Context,
        listener: PurchasesUpdatedListener
    ): BillingClient {
        return BillingClient.newBuilder(context)
            .setListener(listener)
            .enablePendingPurchases()
            .build()
    }
}